src/app/models/ObjectSchemaModel.ts

```diff
@@ -0,0 +1,41 @@
+import 'reflect-metadata';
+import { jsonObject, jsonMember, jsonArrayMember } from 'typedjson';
+
+/**
+ * Object Schema Model is a self refrential field
+ * which is used to model the object schema documented here : 
+ * https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/dtdlv2.md#object
+ * 
+ * This is needed to acommidate json seralization to the jsonLD model. 
+ */
+@jsonObject
+export class ObjectSchemaModel { 
+    @jsonMember
+    public id!: string;
+
+    @jsonMember
+    public name!: string; 
+
+    @jsonMember
+    public description!: string; 
+
+    @jsonMember
+    public displayName!: string; 
+
+    @jsonMember
+    public comment!: string;
+
+    @jsonMember
+    public schema!: string;
+
+    @jsonArrayMember(ObjectSchemaModel) 
+    public fields!: ObjectSchemaModel[];
+
+    public level: number; 
```

Comment:
So if this isn't decorated with `@jsonMember` does that mean this doesn't get serdes? That's good to know if true. Otherwise, this could be an issue when writing out the JSON-LD.

Reply:
So... didnt think about the serdes, but yes it will need to be. I use this to verify the depth of the dtdl object schema. It can be definition be a max of five levels. We could potentially just build it back up we de-seralizing?

Reply:
IDK, I don't see how level would need to be deserialized. You could just re-build the tree and calculate the level and validate during the process. I'm guessing that we should probably allow an import of infinite number of levels if that's how the DTDL is defined but just show a validation error when it's > 5. For no other reason this would at least be beneficial if, in the future, they allow more than 5 levels.